app/services/idv/steps/doc_auth_base_step.rb

```diff
@@ -58,9 +58,11 @@ def idv_failure(result)
       end
 
       def verify_back_image(reset_step:)
-        back_image_verified, data, analytics_hash = assure_id_results
-        data[:notice] = I18n.t('errors.doc_auth.general_info') if data.class == Hash
-        return friendly_failure(data, analytics_hash) unless back_image_verified
+        general_info_message = I18n.t('errors.doc_auth.general_info')
+
+        back_image_verified, data = assure_id_results
+        data[:notice] = general_info_message if data.class == Hash
```

Comment:
personally I'd prefer `.kind_of?` or `.is_a`? In case `HashWithIndifferentAccess` ever sneaks in there?

Reply:
Yeah, ideally `data` should never be a String. I think confusion around that is what is causing us issues. I just pushed a commit to try and back that out.